examples/Extensions/AddSitelinksUsingFeeds.cs

```diff
@@ -0,0 +1,366 @@
+            };
+            
+            feedMapping.AttributeFieldMappings.Add(linkTextAttribute);
+            feedMapping.AttributeFieldMappings.Add(finalUrlAttribute);
+            feedMapping.AttributeFieldMappings.Add(line1Attribute);
+            feedMapping.AttributeFieldMappings.Add(line2Attribute);
+
+            FeedMappingOperation operation = new FeedMappingOperation()
+            {
+                Create = feedMapping
+            };
+            
+            FeedMappingServiceClient feedMappingServiceClient = 
+                client.GetService(Services.V3.FeedMappingService);
+
+            MutateFeedMappingsResponse response = feedMappingServiceClient.MutateFeedMappings
+                (customerId.ToString(), new[] {operation});
+            
+            Console.WriteLine($"Created feed mapping '{response.Results.First().ResourceName}'");
+        }
+
+        private void CreateCampaignFeed(GoogleAdsClient client, long customerId, long campaignId,
+            Hashtable feedData)
+        {
+            List<long> feedItemIds = (List<long>)feedData["feedItemIds"];
+            // Convert all the Feed Item IDs to strings and collapse them into a single
+            // comma-separated string.
+            string aggregatedFeedItemIds = String.Join(',', feedItemIds);
+            CampaignFeed campaignFeed = new CampaignFeed()
+            {
+                Feed = feedData["feed"].ToString(),
+                Campaign = ResourceNames.Campaign(customerId, campaignId),
+                MatchingFunction = new MatchingFunction()
+                {
+                    FunctionString = $"AND(IN(FEED_ITEM_ID,{{ {aggregatedFeedItemIds} }})," +
+                                     "EQUALS(CONTEXT.DEVICE,'Mobile'))"
+                }
+            };
+            campaignFeed.PlaceholderTypes.Add(PlaceholderTypeEnum.Types.PlaceholderType.Sitelink);
+            
+            CampaignFeedOperation operation = new CampaignFeedOperation()
+            {
+                Create = campaignFeed
+            };
+
+            CampaignFeedServiceClient campaignFeedServiceClient = 
+                client.GetService(Services.V3.CampaignFeedService);
+
+            MutateCampaignFeedsResponse response = campaignFeedServiceClient.MutateCampaignFeeds(
+                customerId.ToString(), new[] {operation});
+            
+            Console.WriteLine($"Created campaign feed '{response.Results.First().ResourceName}'");
+        }
+
+        private void CreateAdGroupTargeting(GoogleAdsClient client, long customerId,
+            Hashtable feedData, long adGroupId)
+        {
+            List<string> feedItemResourceNames = (List<string>) feedData["feedItems"];
+            string feedItem = feedItemResourceNames.First();
+            
+            FeedItemTarget feedItemTarget = new FeedItemTarget()
+            {
+                FeedItem = feedItem,
```

Comment:
A comment is missing `You must set targeting on a per-feed-item basis. This will restrict the first feed item we added to only serve for the given ad group.`

Reply:
Added; oversight on my part.